snippets/fuzzySearch.md

```diff
@@ -0,0 +1,33 @@
+### fuzzySearch
+
+Determines if the `patrn` matches with `str`
+
+Loops through `str` and determines if it contains all characters of `patrn` and in the correct order. Both the strings are converted to lower case.
+
+Taken from [here](https://github.com/forrestthewoods/lib_fts/blob/80f3f8c52db53428247e741b9efe2cde9667050c/code/fts_fuzzy_match.js#L18).
+``` js
+fuzzySearch = (patrn, str) => {
+    const pattern = patrn;
```

Comment:
Why is there a re-declaration?

Reply:
By shadow assigning to an equivilant value we can skip declaring the length property statically so we can have fewer lines of code
```js
const fuzzySearch = (pattern, string) => {
  pattern = String(pattern);
  string = String(string);
```

Reply:
Re-declaration is needed to avoid changing the values